Fitness/Models/Habit.swift

```diff
@@ -11,11 +11,40 @@ import Foundation
 
 // Habits are stored by category as a dictionary mapping habit titles to an array of
 // dates on which that habit was checked in [dates are strings of form "MMDDYYYY"].
-// The active habits are stored in an array of form: [ cardio habit, strength habit, mindfulness habit]
+// The active habits are stored as strings in an array of form: [ cardio habit, strength habit, mindfulness habit]
 struct Habit {
     let title: String
-    let dates: [Date]
+    var dates: [Date]
+    let type: HabitTrackingType
     
+    /// Number of consective days going back from today that this habit has been logged;
+    /// includes today if the habit has been logged today
+    var streak: Int {
+        if dates.count == 0 {
+            return 0
+        }
+        
+        var sortedDates = dates.sorted()
+        sortedDates.reverse()
+        
+        if !( sortedDates[0].isToday() || sortedDates[0].isYesterday() ) {
+            return 0
+        }
+        
+        var streak = 1
+
+        for i in 0..<(sortedDates.count-1) {
+            if sortedDates[i] - TimeInterval(floatLiteral: 86400) == sortedDates[i+1] {
```

Comment:
Pull 86400 or `TimeInterval(floatLiteral: 86400)` to top of file. This helps people read the code later because it'll be a name instead.

Reply:
Since I use that constant in other files I'll include it as an extension of `Date`